lld/ELF/ScriptParser.cpp

```diff
@@ -1098,12 +1113,31 @@ SymbolAssignment *ScriptParser::readProvideHidden(bool provide, bool hidden) {
   return cmd;
 }
 
+// Replace whitespace sequence (including \n) with one single space. The output
+// is used by -Map.
+static void squeezeSpaces(std::string &str) {
```

Comment:
I was wondering if there was a way of writing this using algorthim's but I think the end result was worse, it also has the property of removing all leading white-space (which may not be a problem if that's been skipped already).

Best I came up with:
```
  std::replace_if(
      s.begin(), s.end(), [](char c) { return isspace(c); }, ' ');
  s.erase(std::unique(s.begin(), s.end(),
                      [](char lhs, char rhs) {
    return lhs == ' ' && rhs == ' '; }),
          s.end();
```

Reply:
I agree that `<algorithm>` functions might make the algorithm more difficult to read. It will likely require two passes. I've now changed this to a more clever loop:
```cpp
static void squeezeSpaces(std::string &str) {
  char prev = '\0';
  auto it = str.begin();
  for (char c : str)
    if (!isSpace(c) || (c = ' ') != prev)
      *it++ = prev = c;
  str.erase(it, str.end());
}
```